build/make-csv-generator.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -e
+
+if [ -x "${BASEPATH}/../operator-sdk" ]; then
+       OPERATOR_SDK="${BASEPATH}/../operator-sdk"
+else       
+    which operator-sdk &> /dev/null || {
+        echo "operator-sdk not found (see https://github.com/operator-framework/operator-sdk)"
+        exit 1
+    }
+    OPERATOR_SDK="operator-sdk"
+fi
+
+MANIFESTS_GENERATED_DIR="manifests/generated"
+MANIFESTS_GENERATED_CSV=${MANIFESTS_GENERATED_DIR}/kubevirt-ssp-operator.vVERSION.clusterserviceversion.yaml
+PLACEHOLDER_CSV_VERSION="9999.9999.9999"
+
+# Create CSV with placeholder version. The version
+# has to be semver compatible in order for the 
+# operator sdk to create it for us. That's why we
+# are using the absurd 9999.9999.9999 version here. 
+${OPERATOR_SDK} olm-catalog gen-csv --csv-version ${PLACEHOLDER_CSV_VERSION}
+
+# Move CSV to generated folder
+mv deploy/olm-catalog/kubevirt-ssp-operator/${PLACEHOLDER_CSV_VERSION}/kubevirt-ssp-operator.v${PLACEHOLDER_CSV_VERSION}.clusterserviceversion.yaml $MANIFESTS_GENERATED_CSV
+
+# cleanup placeholder version's deployment dir
+rm -rf mv deploy/olm-catalog/kubevirt-ssp-operator/${PLACEHOLDER_CSV_VERSION}
+
+# replace placeholder version with a human readable variable name
+# that will be used later on by csv-generator
+sed -i "s/${PLACEHOLDER_CSV_VERSION}/PLACEHOLDER_CSV_VERSION/g" $MANIFESTS_GENERATED_CSV 
+
+# inject the CRD and Description related data into the CSV
+cp $MANIFESTS_GENERATED_CSV ${MANIFESTS_GENERATED_CSV}.tmp
+./build/update-olm.py ${MANIFESTS_GENERATED_CSV}.tmp > ${MANIFESTS_GENERATED_CSV}
```

Comment:
@fromanirh  is there a chance update-olm.py with this PR change needs to be updated for every version, lets say if make some structural changes or naming ? (asking since you wrote it)

Reply:
> I want to consider this option of putting, say a go template CSV + other OLM manifests, in the container image but "even though I do not think it is the right approach" gives me serious pause. Putting manifests in the container image is specifically the recommendation for OLM installed operators, putting templates on the container image is not a huge deviation from that. 

the image has a templated CSV yaml file that's being injected into it. the csv-generator script is  validating args and doing search/replace on that template, then spitting it to stdout.

Reply:
@yanirq we indeed need to do some changes to the update-olm.py script - but this seems to be needed only when add new CRDs, like when I just did in v1.0.5 and missed that. Nothing else seems to be needed.
If nothing else, this PR is a good chance to review our automation and (admittedly messy) flow and simplify them